### PR TITLE
CLOUDP-183519: introduce custom handling for valid ISO-8601 timestamps

### PR DESCRIPTION
**Pull request by matt-condon**
## Proposed changes

For v1 SDK, timestamps were sent from the SDK as strings and only parsed once they reached the API 

Since migrating to autogenerated v2 Go SDK, timestamps are sent as `time.Time` structs, meaning we need to parse 

The end result is that we are using a different parser since moving to the v2 SDK, which needs to support some custom layouts that were supported by the API for the v1 SDK

_Jira ticket:_ CLOUDP-183519

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by wtrocki** [comment]
> Since migrating to autogenerated v2 Go SDK, timestamps are sent as time.Time structs, meaning we need to parse

FYI - we can change that on the SDK level. This will mean that SDK will not offer any type of date validation and leave users with understanding what date string backend supports.

File: internal/time/time.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package time
+
+import "time"
+
+func ParseTimestamp(timestamp string) (time.Time, error) {
+	layouts := []string{
```
Comment (wtrocki) [inline internal/time/time.go:20]: I think we look for 2 new formats where the timezone is hardcoded 
```
YYYY-MM-DDTHH:MM:SS-0700
YYYY-MM-DDTHH:MM:SSZ07:00
```

In this format timezone is hardcoded. Meaning:
 "0700" indicating the time's offset from UTC in hours and minutes.

We need to generalizethat.. but that leads us to the standard time format (with and without zulu)
image

File: internal/time/time.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		time.RFC3339,
+		"2006-01-02T15:04:05-0700",
```
Comment (wtrocki) [inline internal/time/time.go:22]: This looks like `ISO 8601`

**Review by wtrocki — APPROVED** [review]

File: internal/time/time.go (hunk 2 above)
Comment (wtrocki) [reply internal/time/time.go:22]: This doesn't have representation so we can keep it in such form but I will name it as constant to make it clear.

**Comment by wtrocki** [comment]
Disscussed offline. We do not need to change SDK as we do parse time to string in unified way:
https://github.com/mongodb/atlas-sdk-go/blob/a45c8e06d173e50c7467eb4497adbf3b2b01cf6a/tools/config/go-templates/client.mustache#L172

**Review by wtrocki — DISMISSED** [review]

**Review by swattyT — DISMISSED** [review]
LGTM! Thanks for fixing this!

**Review by swattyT — COMMENTED** [review]
I see `time.Parse()` used at these places as well:
https://github.com/mongodb/mongodb-atlas-cli/blob/6e4fed49212a9480d1f6262b4968910164804af5/internal/convert/database_user.go#L63
https://github.com/mongodb/mongodb-atlas-cli/blob/6e4fed49212a9480d1f6262b4968910164804af5/internal/cli/atlas/datalakepipelines/availablesnapshots/list.go#L59
https://github.com/mongodb/mongodb-atlas-cli/blob/6e4fed49212a9480d1f6262b4968910164804af5/internal/cli/atlas/datalakepipelines/availablesnapshots/list.go#L66

**Review by wtrocki — APPROVED** [review]

**Review by swattyT — APPROVED** [review]
LGTM! Thanks for the changes 🥇

**Comment by wtrocki** [comment]
CC @fmenezes for CLI generation efforts.

**Pull request merged**